File: shifai-android/app/src/main/java/com/shifai/domain/export/MedicalExportEngine.kt
package com.shifai.domain.export

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.pdf.PdfDocument
import java.io.ByteArrayOutputStream
import java.text.SimpleDateFormat
import java.util.*

/**
 * Medical Export PDF Engine — Android
 * S6-1: PDF generation using android.graphics.pdf.PdfDocument
 * S6-2: SOPK template
 * S6-3: Endométriose template
 * S6-4: Custom template
 */
class MedicalExportEngine {

    enum class ExportTemplate(val label: String, val description: String) {
        SOPK("SOPK", "Irrégularité cycles, symptômes androgéniques"),
        ENDOMETRIOSIS("Endométriose", "Douleurs chroniques, localisation, évolution"),
        CUSTOM("Personnalisé", "Sélection libre de sections")
    }

    enum class ExportSection(val label: String) {
        CYCLE_OVERVIEW("Aperçu des cycles"),
        SYMPTOM_FREQUENCY("Fréquence des symptômes"),
        BODY_MAP_HEATMAP("Body Map — Zones de douleur"),
        SLEEP_ENERGY("Patterns sommeil/énergie"),
        CORRELATIONS("Corrélations détectées"),
        PREDICTIONS("Prédictions"),
        MOOD_TIMELINE("Timeline humeur")
    }

    data class ExportConfig(
        val template: ExportTemplate,
        val startDate: Date,
        val endDate: Date,
        val sections: Set<ExportSection>,
        val gynecologistNotes: String? = null
    )

    // A4: 595 x 842 points (72 dpi)
    private val pageWidth = 595
    private val pageHeight = 842
    private val margin = 50f

    fun generatePDF(config: ExportConfig): ByteArray {
        val document = PdfDocument()

        // Page 1: Cover
        var pageNum = 1
        val coverInfo = PdfDocument.PageInfo.Builder(pageWidth, pageHeight, pageNum).create()
        val coverPage = document.startPage(coverInfo)
        drawCoverPage(coverPage.canvas, config)
        document.finishPage(coverPage)

        // Content pages
        for (section in config.sections.sortedBy { it.label }) {
            pageNum++
            val info = PdfDocument.PageInfo.Builder(pageWidth, pageHeight, pageNum).create()
            val page = document.startPage(info)
            drawSection(page.canvas, section, config)
            document.finishPage(page)
        }

        // Disclaimer page
        pageNum++
        val disclaimerInfo = PdfDocument.PageInfo.Builder(pageWidth, pageHeight, pageNum).create()
        val disclaimerPage = document.startPage(disclaimerInfo)
        drawDisclaimerPage(disclaimerPage.canvas)
        document.finishPage(disclaimerPage)

        val output = ByteArrayOutputStream()
        document.writeTo(output)
        document.close()

        return output.toByteArray()
    }

    // MARK: - Cover Page

    private fun drawCoverPage(canvas: Canvas, config: ExportConfig) {
        var y = margin
        val titlePaint = Paint().apply {
            color = Color.rgb(124, 92, 252)
            textSize = 28f
            isFakeBoldText = true
            isAntiAlias = true
        }
        canvas.drawText("Rapport Médical ShifAI", margin, y + 28f, titlePaint)
        y += 50f

        val bodyPaint = Paint().apply {
            color = Color.DKGRAY
            textSize = 12f
            isAntiAlias = true
        }

        val dateFormat = SimpleDateFormat("d MMMM yyyy", Locale.FRANCE)
        canvas.drawText("Template: ${config.template.label}", margin, y, bodyPaint)
        y += 20f
        canvas.drawText(
            "Période: ${dateFormat.format(config.startDate)} — ${dateFormat.format(config.endDate)}",
            margin, y, bodyPaint
        )
        y += 30f

        // Separator
        val linePaint = Paint().apply { color = Color.LTGRAY; strokeWidth = 0.5f }
        canvas.drawLine(margin, y, pageWidth - margin, y, linePaint)
        y += 20f

        val bulletPaint = Paint().apply { color = Color.BLACK; textSize = 13f; isAntiAlias = true }
        val bullets = listOf(
            "📊 Sections incluses: ${config.sections.size}",
            "🔒 Données chiffrées AES-256",
            "⚠️ Ce document est informatif uniquement"
        )
        for (b in bullets) {
            canvas.drawText(b, margin, y, bulletPaint)
            y += 22f
        }

        config.gynecologistNotes?.let { notes ->
            y += 20f
            val notesPaint = Paint().apply { color = Color.BLACK; textSize = 13f; isFakeBoldText = true; isAntiAlias = true }
            canvas.drawText("Notes pour le gynécologue:", margin, y, notesPaint)
            y += 20f
            val notesBody = Paint().apply { color = Color.DKGRAY; textSize = 11f; isAntiAlias = true }
            canvas.drawText(notes, margin, y, notesBody)
        }

        drawWatermark(canvas)
    }

    // MARK: - Section Drawing

    private fun drawSection(canvas: Canvas, section: ExportSection, config: ExportConfig) {
        var y = margin
        val headerPaint = Paint().apply {
            color = Color.rgb(124, 92, 252)
            textSize = 20f
            isFakeBoldText = true
            isAntiAlias = true
        }
        canvas.drawText(section.label, margin, y + 20f, headerPaint)
        y += 50f

        val bodyPaint = Paint().apply { color = Color.BLACK; textSize = 12f; isAntiAlias = true }

        when (section) {
            ExportSection.CYCLE_OVERVIEW -> {
                canvas.drawText("Analyse des cycles sur la période sélectionnée.", margin, y, bodyPaint)
                y += 40f
                // Phase bar
                val phases = listOf(
                    Triple("Menstruel", 0.18f, Color.rgb(239, 68, 68)),
                    Triple("Folliculaire", 0.25f, Color.rgb(52, 211, 153)),
                    Triple("Ovulatoire", 0.14f, Color.rgb(245, 158, 11)),
                    Triple("Lutéal", 0.43f, Color.rgb(167, 139, 250))
                )
                var x = margin
                val barW = pageWidth - 2 * margin
                val barPaint = Paint().apply { isAntiAlias = true }
                for ((label, ratio, color) in phases) {
                    val w = barW * ratio
                    barPaint.color = color
                    barPaint.alpha = 180
                    canvas.drawRect(x, y, x + w, y + 30f, barPaint)
                    val labelPaint = Paint().apply { this.color = Color.WHITE; textSize = 8f; isAntiAlias = true }
                    canvas.drawText(label, x + 4f, y + 18f, labelPaint)
                    x += w
                }
            }
            ExportSection.SYMPTOM_FREQUENCY -> {
                canvas.drawText("Top symptômes les plus fréquents.", margin, y, bodyPaint)
                y += 30f
                val symptoms = listOf("Crampes" to 0.85f, "Fatigue" to 0.72f, "Migraine" to 0.65f,
                    "Ballonnement" to 0.55f, "Anxiété" to 0.48f)
                val barPaint = Paint().apply { isAntiAlias = true }
                for ((name, ratio) in symptoms) {
                    canvas.drawText(name, margin, y + 10f, bodyPaint)
                    barPaint.color = Color.rgb(124, 92, 252)
                    barPaint.alpha = (ratio * 255).toInt()
                    val w = (pageWidth - 2 * margin - 120f) * ratio
                    canvas.drawRect(margin + 110f, y, margin + 110f + w, y + 14f, barPaint)
                    y += 22f
                }
            }
            ExportSection.BODY_MAP_HEATMAP -> {
                canvas.drawText("Zones de douleur signalées.", margin, y, bodyPaint)
            }
            ExportSection.SLEEP_ENERGY -> {
                canvas.drawText("Patterns sommeil/énergie par phase.", margin, y, bodyPaint)
            }
            ExportSection.CORRELATIONS -> {
                canvas.drawText("Corrélations statistiques détectées (r > 0.3).", margin, y, bodyPaint)
            }
            ExportSection.PREDICTIONS -> {
                canvas.drawText("Prédictions et précision historique.", margin, y, bodyPaint)
            }
            ExportSection.MOOD_TIMELINE -> {
                canvas.drawText("Évolution humeur sur la période.", margin, y, bodyPaint)
            }
        }

        drawWatermark(canvas)
    }

    // MARK: - Disclaimer

    private fun drawDisclaimerPage(canvas: Canvas) {
        var y = margin
        val titlePaint = Paint().apply {
            color = Color.RED
            textSize = 18f
            isFakeBoldText = true
            isAntiAlias = true
        }
        canvas.drawText("⚠️ Avertissement Médical", margin, y + 18f, titlePaint)
        y += 50f

        val bodyPaint = Paint().apply { color = Color.BLACK; textSize = 11f; isAntiAlias = true }
        val lines = listOf(
            "Ce document a été généré automatiquement par ShifAI.",
            "Il ne constitue en aucun cas un diagnostic ou avis médical.",
            "Les données sont auto-déclarées et non validées par un professionnel.",
            "",
            "Consultez toujours un professionnel de santé qualifié.",
            "",
            "ShifAI respecte le RGPD. Données chiffrées AES-256, stockées en UE.",
            "",
            "© ShifAI ${Calendar.getInstance().get(Calendar.YEAR)}"
        )
        for (line in lines) {
            canvas.drawText(line, margin, y, bodyPaint)
            y += 18f
        }

        drawWatermark(canvas)
    }

    // MARK: - Watermark

    private fun drawWatermark(canvas: Canvas) {
        val paint = Paint().apply { color = Color.LTGRAY; textSize = 8f; isAntiAlias = true }
        canvas.drawText("Information uniquement — Généré par ShifAI", margin, pageHeight - 20f, paint)
    }

    companion object {
        fun defaultSections(template: ExportTemplate): Set<ExportSection> = when (template) {
            ExportTemplate.SOPK -> setOf(
                ExportSection.CYCLE_OVERVIEW, ExportSection.SYMPTOM_FREQUENCY,
                ExportSection.BODY_MAP_HEATMAP, ExportSection.SLEEP_ENERGY, ExportSection.CORRELATIONS
            )
            ExportTemplate.ENDOMETRIOSIS -> setOf(
                ExportSection.CYCLE_OVERVIEW, ExportSection.BODY_MAP_HEATMAP,
                ExportSection.SYMPTOM_FREQUENCY, ExportSection.SLEEP_ENERGY, ExportSection.MOOD_TIMELINE
            )
            ExportTemplate.CUSTOM -> ExportSection.values().toSet()
        }
    }
}

File: shifai-ios/ShifAI/Domain/Export/MedicalExportEngine.swift
import Foundation
import UIKit
import PDFKit

// MARK: - PDF Generation Engine (S6-1)
// Native PDFKit generation, in-memory, <10s for 3 months data
// S6-2: SOPK template
// S6-3: Endométriose template
// S6-4: Custom template

final class MedicalExportEngine {

    // MARK: - Template Types

    enum ExportTemplate: String, CaseIterable, Identifiable {
        case sopk = "SOPK"
        case endometriosis = "Endométriose"
        case custom = "Personnalisé"

        var id: String { rawValue }

        var description: String {
            switch self {
            case .sopk: return "Irrégularité cycles, symptômes androgéniques, corrélations hormonales"
            case .endometriosis: return "Douleurs chroniques, localisation, intensité, évolution temporelle"
            case .custom: return "Sélection libre de sections"
            }
        }
    }

    struct ExportConfig {
        let template: ExportTemplate
        let dateRange: ClosedRange<Date>
        let sections: Set<ExportSection>
        let gynecologistNotes: String?

        static func defaultConfig(template: ExportTemplate) -> ExportConfig {
            let endDate = Date()
            let startDate = Calendar.current.date(byAdding: .month, value: -3, to: endDate)!
            return ExportConfig(
                template: template,
                dateRange: startDate...endDate,
                sections: template.defaultSections,
                gynecologistNotes: nil
            )
        }
    }

    enum ExportSection: String, CaseIterable, Identifiable {
        case cycleOverview = "Aperçu des cycles"
        case symptomFrequency = "Fréquence des symptômes"
        case bodyMapHeatmap = "Body Map — Zones de douleur"
        case sleepEnergyPatterns = "Patterns sommeil/énergie"
        case correlations = "Corrélations détectées"
        case predictions = "Prédictions"
        case moodTimeline = "Timeline humeur"

        var id: String { rawValue }
    }

    // MARK: - PDF Generation

    func generatePDF(config: ExportConfig) -> Data {
        let pageRect = CGRect(x: 0, y: 0, width: 595, height: 842) // A4
        let renderer = UIGraphicsPDFRenderer(bounds: pageRect)

        return renderer.pdfData { context in
            // Page 1: Cover + Summary
            context.beginPage()
            drawCoverPage(in: context, rect: pageRect, config: config)

            // Page 2+: Sections
            for section in config.sections.sorted(by: { $0.rawValue < $1.rawValue }) {
                context.beginPage()
                drawSection(section, in: context, rect: pageRect, config: config)
            }

            // Final page: Disclaimer
            context.beginPage()
            drawDisclaimerPage(in: context, rect: pageRect)
        }
    }

    // MARK: - Cover Page

    private func drawCoverPage(in ctx: UIGraphicsPDFRendererContext, rect: CGRect, config: ExportConfig) {
        let margin: CGFloat = 50
        var y: CGFloat = margin

        // Logo / Title
        let title = "Rapport Médical ShifAI"
        let titleAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 28, weight: .bold),
            .foregroundColor: UIColor(red: 0.49, green: 0.36, blue: 0.99, alpha: 1)
        ]
        title.draw(at: CGPoint(x: margin, y: y), withAttributes: titleAttrs)
        y += 45

        // Template badge
        let templateText = "Template: \(config.template.rawValue)"
        let templateAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 14, weight: .medium),
            .foregroundColor: UIColor.darkGray
        ]
        templateText.draw(at: CGPoint(x: margin, y: y), withAttributes: templateAttrs)
        y += 25

        // Date range
        let dateFormatter = DateFormatter()
        dateFormatter.dateStyle = .long
        dateFormatter.locale = Locale(identifier: "fr_FR")
        let rangeText = "Période: \(dateFormatter.string(from: config.dateRange.lowerBound)) — \(dateFormatter.string(from: config.dateRange.upperBound))"
        let rangeAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 12),
            .foregroundColor: UIColor.gray
        ]
        rangeText.draw(at: CGPoint(x: margin, y: y), withAttributes: rangeAttrs)
        y += 30

        // Separator
        let path = UIBezierPath()
        path.move(to: CGPoint(x: margin, y: y))
        path.addLine(to: CGPoint(x: rect.width - margin, y: y))
        UIColor.lightGray.setStroke()
        path.lineWidth = 0.5
        path.stroke()
        y += 20

        // Summary metrics
        let summaryAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 14),
            .foregroundColor: UIColor.black
        ]
        let metrics = [
            "📊 Sections incluses: \(config.sections.count)",
            "📅 Durée couverte: \(daysBetween(config.dateRange.lowerBound, config.dateRange.upperBound)) jours",
            "🔒 Données chiffrées AES-256 — déchiffrées localement pour export",
            "⚠️ Ce document est informatif uniquement"
        ]

        for metric in metrics {
            metric.draw(at: CGPoint(x: margin, y: y), withAttributes: summaryAttrs)
            y += 22
        }

        // Gynecologist notes
        if let notes = config.gynecologistNotes, !notes.isEmpty {
            y += 20
            let notesTitle = "Notes pour le gynécologue:"
            let notesTitleAttrs: [NSAttributedString.Key: Any] = [
                .font: UIFont.systemFont(ofSize: 14, weight: .semibold),
                .foregroundColor: UIColor.black
            ]
            notesTitle.draw(at: CGPoint(x: margin, y: y), withAttributes: notesTitleAttrs)
            y += 22

            let notesAttrs: [NSAttributedString.Key: Any] = [
                .font: UIFont.systemFont(ofSize: 12),
                .foregroundColor: UIColor.darkGray
            ]
            let notesRect = CGRect(x: margin, y: y, width: rect.width - 2 * margin, height: 200)
            notes.draw(in: notesRect, withAttributes: notesAttrs)
        }

        // Watermark
        drawWatermark(in: rect)
    }

    // MARK: - Section Drawing

    private func drawSection(_ section: ExportSection, in ctx: UIGraphicsPDFRendererContext, rect: CGRect, config: ExportConfig) {
        let margin: CGFloat = 50
        var y: CGFloat = margin

        // Section header
        let headerAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 20, weight: .bold),
            .foregroundColor: UIColor(red: 0.49, green: 0.36, blue: 0.99, alpha: 1)
        ]
        section.rawValue.draw(at: CGPoint(x: margin, y: y), withAttributes: headerAttrs)
        y += 35

        let bodyAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 12),
            .foregroundColor: UIColor.black
        ]

        switch section {
        case .cycleOverview:
            drawCycleOverview(at: &y, margin: margin, width: rect.width, attrs: bodyAttrs, config: config)
        case .symptomFrequency:
            drawSymptomFrequency(at: &y, margin: margin, width: rect.width, attrs: bodyAttrs, config: config)
        case .bodyMapHeatmap:
            drawBodyMapHeatmap(at: &y, margin: margin, width: rect.width, config: config)
        case .sleepEnergyPatterns:
            drawSleepEnergyPatterns(at: &y, margin: margin, width: rect.width, attrs: bodyAttrs, config: config)
        case .correlations:
            drawCorrelations(at: &y, margin: margin, width: rect.width, attrs: bodyAttrs, config: config)
        case .predictions:
            drawPredictions(at: &y, margin: margin, width: rect.width, attrs: bodyAttrs, config: config)
        case .moodTimeline:
            drawMoodTimeline(at: &y, margin: margin, width: rect.width, attrs: bodyAttrs, config: config)
        }

        drawWatermark(in: rect)
    }

    // MARK: - S6-1: Cycle Overview Chart

    private func drawCycleOverview(at y: inout CGFloat, margin: CGFloat, width: CGFloat, attrs: [NSAttributedString.Key: Any], config: ExportConfig) {
        let text = """
        Ce rapport couvre \(daysBetween(config.dateRange.lowerBound, config.dateRange.upperBound)) jours de suivi.

        Les données ci-dessous proviennent de votre journal quotidien ShifAI.
        Chaque cycle est identifié par le début de vos règles.

        Longueur moyenne des cycles: — (calculée à partir des données)
        Écart-type: — (régularité du cycle)
        Tendance: — (stable / raccourcissement / allongement)
        """
        let textRect = CGRect(x: margin, y: y, width: width - 2 * margin, height: 200)
        text.draw(in: textRect, withAttributes: attrs)
        y += 160

        // Cycle timeline bar placeholder
        let barRect = CGRect(x: margin, y: y, width: width - 2 * margin, height: 40)
        let phases: [(color: UIColor, width: CGFloat, label: String)] = [
            (.systemRed.withAlphaComponent(0.6), 0.18, "Menstruel"),
            (.systemGreen.withAlphaComponent(0.4), 0.25, "Folliculaire"),
            (.systemOrange.withAlphaComponent(0.5), 0.14, "Ovulatoire"),
            (.systemPurple.withAlphaComponent(0.4), 0.43, "Lutéal")
        ]

        var x = margin
        for phase in phases {
            let w = (width - 2 * margin) * phase.width
            let rect = CGRect(x: x, y: y, width: w, height: 30)
            phase.color.setFill()
            UIBezierPath(roundedRect: rect, cornerRadius: 4).fill()

            let labelAttrs: [NSAttributedString.Key: Any] = [
                .font: UIFont.systemFont(ofSize: 8, weight: .medium),
                .foregroundColor: UIColor.white
            ]
            phase.label.draw(at: CGPoint(x: x + 4, y: y + 10), withAttributes: labelAttrs)
            x += w
        }
        y += 50
    }

    // MARK: - S6-2: Symptom Frequency (SOPK Top 10)

    private func drawSymptomFrequency(at y: inout CGFloat, margin: CGFloat, width: CGFloat, attrs: [NSAttributedString.Key: Any], config: ExportConfig) {
        let text = """
        Top 10 symptômes les plus fréquents sur la période sélectionnée.
        Les barres représentent la fréquence relative de chaque symptôme.

        (Les données sont lues depuis la base locale chiffrée)
        """
        let textRect = CGRect(x: margin, y: y, width: width - 2 * margin, height: 80)
        text.draw(in: textRect, withAttributes: attrs)
        y += 80

        // Horizontal bar chart placeholder for top symptoms
        let symptoms = [
            ("Crampes", 0.85), ("Fatigue", 0.72), ("Migraine", 0.65),
            ("Ballonnement", 0.55), ("Anxiété", 0.48), ("Insomnie", 0.40),
            ("Mal de dos", 0.38), ("Acné", 0.30), ("Nausée", 0.25),
            ("Irritabilité", 0.22)
        ]

        let barAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 10),
            .foregroundColor: UIColor.darkGray
        ]

        for (label, ratio) in symptoms {
            label.draw(at: CGPoint(x: margin, y: y), withAttributes: barAttrs)
            let barW = (width - 2 * margin - 100) * CGFloat(ratio)
            let barRect = CGRect(x: margin + 100, y: y + 2, width: barW, height: 12)
            UIColor(red: 0.49, green: 0.36, blue: 0.99, alpha: CGFloat(ratio)).setFill()
            UIBezierPath(roundedRect: barRect, cornerRadius: 3).fill()
            y += 20
        }
    }

    // MARK: - S6-3: Body Map Heatmap (Endo)

    private func drawBodyMapHeatmap(at y: inout CGFloat, margin: CGFloat, width: CGFloat, config: ExportConfig) {
        let titleAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 12),
            .foregroundColor: UIColor.black
        ]

        let text = "Zones de douleur les plus fréquentes. L'intensité de couleur représente la fréquence des signalements."
        text.draw(at: CGPoint(x: margin, y: y), withAttributes: titleAttrs)
        y += 25

        // Simplified body map zones with intensity
        let zones: [(name: String, x: CGFloat, yOff: CGFloat, w: CGFloat, h: CGFloat, intensity: CGFloat)] = [
            ("Utérus", 0.4, 0.35, 0.2, 0.12, 0.8),
            ("Ovaire G", 0.3, 0.38, 0.08, 0.06, 0.5),
            ("Ovaire D", 0.62, 0.38, 0.08, 0.06, 0.6),
            ("Dos", 0.38, 0.15, 0.24, 0.15, 0.7),
            ("Cuisses", 0.3, 0.55, 0.4, 0.1, 0.3)
        ]

        let mapW = width - 2 * margin
        let mapH: CGFloat = 250

        // Body outline
        let outlineRect = CGRect(x: margin + mapW * 0.25, y: y, width: mapW * 0.5, height: mapH)
        UIColor.systemGray5.setFill()
        UIBezierPath(roundedRect: outlineRect, cornerRadius: 16).fill()
        UIColor.systemGray3.setStroke()
        UIBezierPath(roundedRect: outlineRect, cornerRadius: 16).stroke()

        // Pain zones
        for zone in zones {
            let zoneRect = CGRect(
                x: margin + mapW * zone.x,
                y: y + mapH * zone.yOff,
                width: mapW * zone.w,
                height: mapH * zone.h
            )
            UIColor.systemRed.withAlphaComponent(zone.intensity * 0.7).setFill()
            UIBezierPath(roundedRect: zoneRect, cornerRadius: 6).fill()

            let labelAttrs: [NSAttributedString.Key: Any] = [
                .font: UIFont.systemFont(ofSize: 8, weight: .medium),
                .foregroundColor: UIColor.white
            ]
            zone.name.draw(at: CGPoint(x: zoneRect.minX + 2, y: zoneRect.minY + 2), withAttributes: labelAttrs)
        }

        y += mapH + 20
    }

    // MARK: - Sleep/Energy Patterns

    private func drawSleepEnergyPatterns(at y: inout CGFloat, margin: CGFloat, width: CGFloat, attrs: [NSAttributedString.Key: Any], config: ExportConfig) {
        let text = "Patterns de sommeil et d'énergie par phase du cycle."
        text.draw(at: CGPoint(x: margin, y: y), withAttributes: attrs)
        y += 25

        // Table header
        let phases = ["Menstruel", "Folliculaire", "Ovulatoire", "Lutéal"]
        let colW = (width - 2 * margin) / 5

        let headerAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 10, weight: .bold),
            .foregroundColor: UIColor.black
        ]
        let cellAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 10),
            .foregroundColor: UIColor.darkGray
        ]

        // Headers
        "Phase".draw(at: CGPoint(x: margin, y: y), withAttributes: headerAttrs)
        "Sommeil moy".draw(at: CGPoint(x: margin + colW, y: y), withAttributes: headerAttrs)
        "Énergie moy".draw(at: CGPoint(x: margin + colW * 2, y: y), withAttributes: headerAttrs)
        "Stress moy".draw(at: CGPoint(x: margin + colW * 3, y: y), withAttributes: headerAttrs)
        "Humeur".draw(at: CGPoint(x: margin + colW * 4, y: y), withAttributes: headerAttrs)
        y += 18

        // Separator
        let sep = UIBezierPath()
        sep.move(to: CGPoint(x: margin, y: y))
        sep.addLine(to: CGPoint(x: width - margin, y: y))
        UIColor.lightGray.setStroke()
        sep.lineWidth = 0.5
        sep.stroke()
        y += 5

        // Data rows (placeholder)
        let rows = [
            ("Menstruel", "6.5h ⭐3", "3.5/10", "5/5", "😔"),
            ("Folliculaire", "7.2h ⭐4", "7/10", "2/5", "😊"),
            ("Ovulatoire", "7.0h ⭐4", "8.5/10", "2/5", "😄"),
            ("Lutéal", "6.8h ⭐3", "5/10", "4/5", "😐")
        ]

        for row in rows {
            row.0.draw(at: CGPoint(x: margin, y: y), withAttributes: cellAttrs)
            row.1.draw(at: CGPoint(x: margin + colW, y: y), withAttributes: cellAttrs)
            row.2.draw(at: CGPoint(x: margin + colW * 2, y: y), withAttributes: cellAttrs)
            row.3.draw(at: CGPoint(x: margin + colW * 3, y: y), withAttributes: cellAttrs)
            row.4.draw(at: CGPoint(x: margin + colW * 4, y: y), withAttributes: cellAttrs)
            y += 18
        }
    }

    // MARK: - Correlations Table

    private func drawCorrelations(at y: inout CGFloat, margin: CGFloat, width: CGFloat, attrs: [NSAttributedString.Key: Any], config: ExportConfig) {
        let text = "Corrélations statistiques significatives détectées par l'algorithme ShifAI (r > 0.3)."
        text.draw(at: CGPoint(x: margin, y: y), withAttributes: attrs)
        y += 25
    }

    // MARK: - Predictions

    private func drawPredictions(at y: inout CGFloat, margin: CGFloat, width: CGFloat, attrs: [NSAttributedString.Key: Any], config: ExportConfig) {
        let text = "Prédictions générées par le moteur d'intelligence ShifAI et leur précision historique."
        text.draw(at: CGPoint(x: margin, y: y), withAttributes: attrs)
        y += 25
    }

    // MARK: - Mood Timeline

    private func drawMoodTimeline(at y: inout CGFloat, margin: CGFloat, width: CGFloat, attrs: [NSAttributedString.Key: Any], config: ExportConfig) {
        let text = "Évolution de l'humeur sur la période sélectionnée."
        text.draw(at: CGPoint(x: margin, y: y), withAttributes: attrs)
        y += 25
    }

    // MARK: - Disclaimer Page (FR21)

    private func drawDisclaimerPage(in ctx: UIGraphicsPDFRendererContext, rect: CGRect) {
        let margin: CGFloat = 50
        var y: CGFloat = margin

        let titleAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 18, weight: .bold),
            .foregroundColor: UIColor.systemRed
        ]
        "⚠️ Avertissement Médical".draw(at: CGPoint(x: margin, y: y), withAttributes: titleAttrs)
        y += 35

        let disclaimerAttrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 12),
            .foregroundColor: UIColor.black
        ]

        let disclaimer = """
        Ce document a été généré automatiquement par ShifAI à titre informatif uniquement.

        Il ne constitue en aucun cas un diagnostic médical, un avis médical, ou un traitement.
        Les données présentées sont auto-déclarées par l'utilisatrice et n'ont pas été validées
        par un professionnel de santé.

        Les prédictions et corrélations détectées sont basées sur des algorithmes statistiques
        et des modèles d'intelligence artificielle entraînés sur des données agrégées.
        Elles ne doivent pas être utilisées comme base unique de décision médicale.

        Consultez toujours un professionnel de santé qualifié pour toute question médicale.

        ShifAI respecte le RGPD et les réglementations européennes en matière de protection
        des données personnelles de santé. Toutes les données sont chiffrées AES-256, stockées
        en Union Européenne, et ne sont jamais partagées avec des tiers.

        © ShifAI \(Calendar.current.component(.year, from: Date())) — Tous droits réservés
        """

        let disclaimerRect = CGRect(x: margin, y: y, width: rect.width - 2 * margin, height: 600)
        disclaimer.draw(in: disclaimerRect, withAttributes: disclaimerAttrs)

        drawWatermark(in: rect)
    }

    // MARK: - Watermark

    private func drawWatermark(in rect: CGRect) {
        let watermark = "Information uniquement — Généré par ShifAI"
        let attrs: [NSAttributedString.Key: Any] = [
            .font: UIFont.systemFont(ofSize: 8),
            .foregroundColor: UIColor.lightGray
        ]
        watermark.draw(at: CGPoint(x: 50, y: rect.height - 30), withAttributes: attrs)
    }

    // MARK: - Helpers

    private func daysBetween(_ start: Date, _ end: Date) -> Int {
        Calendar.current.dateComponents([.day], from: start, to: end).day ?? 0
    }
}

// MARK: - Template Default Sections

extension MedicalExportEngine.ExportTemplate {
    var defaultSections: Set<MedicalExportEngine.ExportSection> {
        switch self {
        case .sopk:
            return [.cycleOverview, .symptomFrequency, .bodyMapHeatmap,
                    .sleepEnergyPatterns, .correlations]
        case .endometriosis:
            return [.cycleOverview, .bodyMapHeatmap, .symptomFrequency,
                    .sleepEnergyPatterns, .moodTimeline]
        case .custom:
            return Set(MedicalExportEngine.ExportSection.allCases)
        }
    }
}
